File: core/src/pro/superkostya/MainScreen.java
package pro.superkostya;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.maps.tiled.TiledMap;
import com.badlogic.gdx.maps.tiled.TiledMapTileLayer;
import com.badlogic.gdx.maps.tiled.TmxMapLoader;
import com.badlogic.gdx.maps.tiled.renderers.OrthogonalTiledMapRenderer;
import com.badlogic.gdx.scenes.scene2d.Stage;

public class MainScreen implements Screen {
    Stage stage;
    TiledMap map;
    OrthogonalTiledMapRenderer renderer;
    OrthographicCamera camera = new OrthographicCamera();
    Koala koala;

    public void show() {
        map = new TmxMapLoader().load("level1.tmx");
        final float pixelsPerTile = 16;
        renderer = new OrthogonalTiledMapRenderer(map, 1 / pixelsPerTile);

        stage = new Stage();
        stage.getViewport().setCamera(camera);

        koala = new Koala();
        koala.layer = (TiledMapTileLayer) map.getLayers().get("walls");
        koala.setPosition(20, 10);
        stage.addActor(koala);
    }

    public void render(float delta) {
        Gdx.gl.glClearColor(0.5f, 0.5f, 1, 1);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);

        camera.position.x = koala.getX();
        camera.update();

        renderer.setView(camera);
        renderer.render();

        stage.act(delta);
        stage.draw();
    }

    public void dispose() {
    }

    public void hide() {
    }

    public void pause() {
    }

    public void resize(int width, int height) {
        camera.setToOrtho(false, 20 * width / height, 20);
    }

    public void resume() {
    }
}

File: core/src/pro/superkostya/screen/MainScreen.kt
package pro.superkostya.screen

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.glutils.ShapeRenderer
import pro.superkostya.CAMERA_HEIGHT
import pro.superkostya.CAMERA_WIDTH
import pro.superkostya.world.BaseWorld
import pro.superkostya.world.DrawingWorld

class MainScreen : Screen {

    private val camera = OrthographicCamera(CAMERA_WIDTH, CAMERA_HEIGHT)

    private var world: BaseWorld = DrawingWorld()

    private val spriteBatch = SpriteBatch()

    private val debugRenderer = ShapeRenderer()

    override fun show() {
        camera.position.set(CAMERA_WIDTH / 2f, CAMERA_HEIGHT / 2f, 0f)
        Gdx.input.inputProcessor = world
    }

    override fun resize(width: Int, height: Int) {
    }

    override fun resume() {
    }

    override fun render(delta: Float) {
        val color = world.backgroundColor
        Gdx.gl.glClearColor(color.r, color.g, color.b, color.a)
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
        camera.update()
        world.update(delta)
        world.render()
    }

    override fun hide() {
        Gdx.input.inputProcessor = null
    }

    override fun pause() {
    }

    override fun dispose() {
    }
}
